pkg/auth/spiffe_authorizer.go

```diff
@@ -0,0 +1,81 @@
+package auth
+
+import (
+	"context"
+	"path"
+
+	"github.com/buildbarn/bb-storage/pkg/digest"
+	pb "github.com/buildbarn/bb-storage/pkg/proto/configuration/auth"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/go-spiffe/v2/svid/x509svid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+)
+
+type SpiffeAuthorizer struct {
+	*pb.SpiffeAuthorizer
+}
+
+func NewSpiffeAuthorizer(config *pb.AuthorizerConfiguration) Authorizer {
+	spifAuth := config.GetSpiffe()
+	if spifAuth == nil {
+		return &SpiffeAuthorizer{spifAuth}
+	}
+	return &SpiffeAuthorizer{}
+}
+
+func (s *SpiffeAuthorizer) Authorize(ctx context.Context, instanceNames []digest.InstanceName) []error {
```

Comment:
Nobody calls this yet, so it's hard to evaluate if we need to return an array of errors or just one.  Will the caller need to see which of the instance names match?  What happens if more than one match?

Reply:
>  Authorize returns a slice of errors, in the same order as the
> passed instance names.
>
> For each error, a nil value indicates that an instance name was
> authorized.
> A non-nil value indicates that the instance name was not authorized,
> or that an error occurred when authorizing.
>
> Note that this function may block, and should not be called while
> locks are held which may be contended.


https://pkg.go.dev/github.com/buildbarn/bb-storage@v0.0.0-20210909130924-6c0280c9d0f2/pkg/auth#Authorizer